checker/check_runner_test.go

```diff
@@ -0,0 +1,222 @@
+			args: args{
+				checkName: "TestNewRunner2",
+				repo:      "TestNewRunner2",
+				checkReq:  &CheckRequest{},
+			},
+			want: &Runner{
+				CheckName:    "TestNewRunner2",
+				Repo:         "TestNewRunner2",
+				CheckRequest: CheckRequest{},
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // capture range variable
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			if got := NewRunner(tt.args.checkName, tt.args.repo, tt.args.checkReq); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewRunner() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestRunner_SetCheckName(t *testing.T) {
+	t.Parallel()
+	type fields struct {
+		CheckName    string
+		Repo         string
+		CheckRequest CheckRequest
+	}
+	type args struct {
+		check string
+	}
+	tests := []struct { //nolint:govet
+		name   string
+		fields fields
+		args   args
+	}{
+		{
+			name: "TestRunner_SetCheckName",
+			fields: fields{
+				CheckName:    "TestRunner_SetCheckName",
+				Repo:         "TestRunner_SetCheckName",
+				CheckRequest: CheckRequest{},
+			},
+			args: args{
+				check: "TestRunner_SetCheckName",
+			},
+		},
+		{
+			name: "TestRunner_SetCheckName2",
+			fields: fields{
+				CheckName:    "TestRunner_SetCheckName2",
+				Repo:         "TestRunner_SetCheckName2",
+				CheckRequest: CheckRequest{},
+			},
+			args: args{
+				check: "TestRunner_SetCheckName2",
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // capture range variable
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			r := &Runner{
+				CheckName:    tt.fields.CheckName,
+				Repo:         tt.fields.Repo,
+				CheckRequest: tt.fields.CheckRequest,
+			}
+			r.SetCheckName(tt.args.check)
+			if r.CheckName != tt.args.check {
+				t.Errorf("SetCheckName() = %v, want %v", r.CheckName, tt.args.check)
+			}
+		})
+	}
+}
+
+func TestRunner_SetRepo(t *testing.T) {
+	t.Parallel()
+	type fields struct {
+		CheckName    string
+		Repo         string
+		CheckRequest CheckRequest
+	}
+	type args struct {
+		repo string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+	}{
+		{
+			name: "TestRunner_SetCheckName",
+			fields: fields{
+				CheckName:    "TestRunner_SetCheckName",
+				Repo:         "TestRunner_SetCheckName",
+				CheckRequest: CheckRequest{},
+			},
+			args: args{
+				repo: "TestRunner_SetCheckName",
+			},
+		},
+		{
+			name: "TestRunner_SetCheckName2",
+			fields: fields{
+				CheckName:    "TestRunner_SetCheckName2",
+				Repo:         "TestRunner_SetCheckName2",
+				CheckRequest: CheckRequest{},
+			},
+			args: args{
+				repo: "TestRunner_SetCheckName2",
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // capture range variable
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			r := &Runner{
+				CheckName:    tt.fields.CheckName,
+				Repo:         tt.fields.Repo,
+				CheckRequest: tt.fields.CheckRequest,
+			}
+			r.SetRepo(tt.args.repo)
+			if r.Repo != tt.args.repo {
```

Comment:
same comment. if we're setting the repo field to begin with, how do we know SetRepo does anything?